Joss paper (#84)
TUDelftGeodesy/stmtools: paper/paper.md

---
title: 'STMTools: Xarray extension for Interferometric SAR data in Space-Time Matrix format'
tags:
  - Python
  - Interferometric
  - Synthetic Aperture Radar
  - InSAR
  - Dask
  - Xarray
authors:
  - name: Ou Ku
    orcid: 0000-0002-5373-5209
    affiliation: 1 
  - name: Fakhereh Alidoost
    orcid: 0000-0001-8407-6472
    affiliation: 1
  - name: Pranav Chandramouli
    orcid: 0000-0002-7896-2969
    affiliation: 1
  - name: Thijs van Lankveld
    orcid: 0009-0001-1147-4813
    affiliation: 1
  - name: Meiert W. Grootes
    orcid: 0000-0002-5733-4795
    affiliation: 1
  - name: Francesco Nattino
    orcid: 0000-0003-3286-0139
    affiliation: 1
  - name: Freek van Leijen
    orcid: 0000-0002-2582-9267
    affiliation: 2
affiliations:
 - name: Netherlands eScience Center, Netherlands
   index: 1
 - name: Delft University of Technology, Netherlands
   index: 2
date: 23 Dec 2024
bibliography: paper.bib
---

## Summary

> Interferometry Synthetic Aperture Radar (InSAR) is a crucial technology for monitoring ground surface deformation. It has essential value in various applications, such as civil-infrastructure stability [@chang2014detection; @chang2017railway], hydrocarbons extraction [@fokker2016application; @ZHANG2022102847], etc. To efficiently process and analyze these datasets, researchers has proposed the Space-Time Matrix (STM) format for InSAR datasets [@Bruna2021, @vanLeijen2021], which enanbles the intergration of the contextual information with the InSAR data to reveal the mechanisms driving deformation.  

Reviewer: ```suggestion
Interferometry Synthetic Aperture Radar (InSAR) is a commonly used technology for monitoring ground surface deformation in various applications, such as civil-infrastructure stability [@chang2014detection; @chang2017railway], hydrocarbons extraction [@fokker2016application; @ZHANG2022102847]. To process and analyze InSAR datasets, researchers have proposed the Space-Time Matrix (STM) format [@Bruna2021, @vanLeijen2021], which enables the integration of contextual information with data to reveal the mechanisms driving deformation.  
```

Reviewer: Here are explanations about my suggestions:

- I suggest rephrasing "a crucial technology", "essential value" and "To efficiently process" because they rely on relative terms that lack clarity without specific context. For example if we say "efficiently", we should explain what we mean, how efficient it is, and how it is measured. 
- I also suggest removing "etc" as it does not add much to the content.
- I replaced "these datasets" with "InSAR datasets" since "these" was unclear and was not mentioned in the previous sentence. 
- The rest of my suggestions are about fixing typos.

## Statement of Need

> Typiclally, modern time-series InSAR methods is able to provides millions of observation points in a single dataset. However, due the complex and ambiguous nature of InSAR observations, interpretation of these datasets is challenging. [@hanssen2001radar] Under the STM framework, contextual information such as temperature, precipitation, land-use, etc. can be integrated with InSAR data. This can enable a better understanding on the driving mechanisms of the ground deformation, resulting in more reliable and accurate interpretation of the InSAR data. [@Bruna2021, @vanLeijen2021]

Reviewer: ```suggestion
Modern time-series InSAR methods provide millions of observation points in a single dataset. However, interpretation of these datasets is challenging due to the complex and ambiguous nature of InSAR observations. [@hanssen2001radar] Under STM format, contextual information such as temperature, precipitation, and land-use can be integrated with InSAR data. This facilitates a better interpretation of InSAR data, resulting in a reliable and accurate understanding of the mechanisms of ground deformation. [@Bruna2021, @vanLeijen2021]
```

Reviewer: Here is the explanation of my suggestion: a better interpretation of the InSAR data will lead to a better understanding of ground deformation, not the other way around.

> To implement the STM format in Python, we developed the `STMTools` package. `STMTools` is developed as an extension of `Xarray`, leveraging `Xarray`'s support for labeled multi-dimensional arrays for the Space-Time concept. `STMTools` provides a set of tools to efficiently connect the InSAR data with various contextual information, such as cadastral data, weather data, etc. The package also utilizes `Dask` for parallel computing, enabling the processing of large-scale InSAR datasets.

Reviewer: ```suggestion
To utilize the STM format for InSAR data, we developed the `STMTools` package in Python-- as an extension of `Xarray`-- leveraging `Xarray`'s support for labeled multi-dimensional arrays for the Space-Time concept. `STMTools` provides a set of tools to connect InSAR data with various contextual information, such as cadastral data, and weather data. The package also utilizes `Dask` for parallel computing, enabling the efficient processing of large-scale data.
```

Reviewer: Can we also add one sentence about other functionalities of the STMTools package here, for example, subsetting data, adding metadata, regulating the dimensions, ....

> To implement the STM format in Python, we developed the `STMTools` package. `STMTools` is developed as an extension of `Xarray`, leveraging `Xarray`'s support for labeled multi-dimensional arrays for the Space-Time concept. `STMTools` provides a set of tools to efficiently connect the InSAR data with various contextual information, such as cadastral data, weather data, etc. The package also utilizes `Dask` for parallel computing, enabling the processing of large-scale InSAR datasets.

Reviewer 2: - I have added a suggestion on the usage of Xarray datasets for contextual information - please change it if incorrect.
- On the use of Dask, does STMTools uses it directly or only via Xarray? If the former, please drop my suggestion on this topic.
```suggestion
To facilitate the analysis of InSAR datasets following the STM formalism in Python, we developed the `STMTools` package. `STMTools` is developed as an extension of the `Xarray` package, leveraging its support for labeled multi-dimensional arrays for the Space-Time dimensions. `STMTools` provides a set of tools to efficiently connect the InSAR data with various contextual information, such as cadastral data, weather data, etc. The Xarray `Dataset` data structure is used to group InSAR data and the contextual information under shared dimension coordinates (space and/or time). By building on Xarray, STMTools can also leverage `Dask` for parallel computing, enabling the processing of large-scale InSAR datasets.
```

> ## Tutorial

Reviewer 2: Maybe you could call this section as "Main functionalities", with a short list of the main functionalities of STMTools (each with a link to its tutorial).

> We provide a tutorial as a Jupyter notebook to demonstrate the basic functionalities of `STMTools`:

Reviewer: ```suggestion
We provided several tutorials, also available as Jupyter notebooks, to demonstrate the functionalities of `STMTools`:
```

- [Load InSAR data in STM format](https://tudelftgeodesy.github.io/stmtools/stm_init/)

- [Basic operations with an STM](https://tudelftgeodesy.github.io/stmtools/operations/)

> - [Reorder STM in Morton Ordering](https://tudelftgeodesy.github.io/stmtools/order/)

Reviewer: The links provided here, lead to documentation, not Jupyter Notebooks. If the idea is to share the tutorials as notebooks, we can add a download button on the MkDocs page so users can get the notebooks too.

## Acknowledgements

> The authors express sincere gratitude to the Dutch Research Council (Nederlandse Organisatie voor Wetenschappelijk Onderzoek, NWO) for their generous funding of the `STMTools` development through the Collaboration in Innovative Technologies (CIT 2021) Call, grant NLESC.CIT.2021.006. Special thanks to SURF for providing valuable computational resources for `STMTools` testing.

Reviewer: ```suggestion
The authors express sincere gratitude to the Dutch Research Council (Nederlandse Organisatie voor Wetenschappelijk Onderzoek, NWO) for their generous funding of the `STMTools` development through the Collaboration in Innovative Technologies (CIT 2021) Call, grant NLESC.CIT.2021.006. Special thanks to SURF for providing computational resources for `STMTools` testing.
```

Reviewer: Regarding SURF acknowledgement, if the project has received a grant from SURF for using resources, the Acknowledgement text is mentioned in the grant agreement. Please use that text here.
